core/src/main/kotlin/net/corda/core/crypto/CompositeKey.kt

```
@@ -201,8 +212,8 @@ class CompositeKey private constructor(val threshold: Int, children: List<NodeAn
     fun isFulfilledBy(keysToCheck: Iterable<PublicKey>): Boolean {
         // We validate keys only when checking if they're matched, as this checks subkeys as a result.
         // Doing these checks at deserialization/construction time would result in duplicate checks.
-        if (!validated)
-            checkValidity() // TODO: remove when checkValidity() will be eventually invoked during/after deserialization.
+        checkValidity()
+        if (keysToCheck.any { it is CompositeKey }) return false
```

maybe better to throw an exception instead of false here

---

That would be breaking existing behaviour.

---

ok, maybe we should log a warning then?

---

If we find that returning false here was the wrong decision, and apps would benefit from it throwing an exception instead, then we should create a second method which is identical to this except that it throws. But I don't see any benefit of doing that now, not until we have developer feedback.